linode/resource_linode_lke_cluster.go

```diff
@@ -406,6 +414,150 @@ func reconcileLKEClusterPoolSpecs(poolSpecs []linodeLKEClusterPoolSpec, pools []
 	return
 }
 
+func waitForClusterPoolReady(ctx context.Context, client *linodego.Client, errCh chan<- error, wg *sync.WaitGroup, clusterID, poolID int) {
+	eventTicker := time.NewTicker(500 * time.Millisecond)
+
+	for {
+		select {
+		case <-ctx.Done():
+			log.Printf("[ERROR] timed out waiting for LKE Cluster (%d) Pool (%d) to be ready", clusterID, poolID)
+			return
+
+		case <-eventTicker.C:
+			pool, err := client.GetLKEClusterPool(ctx, clusterID, poolID)
+			if err != nil {
+				errCh <- fmt.Errorf("failed to get LKE Cluster (%d) Pool (%d): %w", clusterID, poolID, err)
+			}
+
+			ready := true
+			for _, instance := range pool.Linodes {
+				if instance.Status == linodego.LKELinodeNotReady {
+					ready = false
+				}
```

Comment:
as soon as you hit one, you can stop looping. Maybe instead of ready bool, just bring that logic from below right here and return?

Reply:
I discussed the acceptance criteria for upgrading a cluster with Tim and he said we need to ensure every cluster instance is ready, in case it fails. So I have to wait for all instance IDs to be ready.